prov/efa/src/efa_domain.c

```diff
@@ -273,9 +281,12 @@ int efa_domain_open(struct fid_fabric *fabric_fid, struct fi_info *info,
 		}
 	}
 
-	efa_mr_cache_enable = 0;
-
+	free(domain->cache);
```

Comment:
I think this should be refactored, let's just not allocate this if there is not going to be a cache.

Reply:
Doesn't work that way, we need to allocate before we call ofi_mr_cache_init. If the init fails, we need to free the cache.

Reply:
I refactored it a tad, added an int which is checked, could be a bool but not that important.

Reply:
Oh, right, forgot about the fall through here. A comment might be helpful here but not required for this PR.